classes/Hook/HookActionCarrierProcess.php

```diff
@@ -52,17 +53,35 @@ public function run()
             );
 
             $carrierName = $carrierRepository->findByCarrierId((int) $this->params['cart']->id_carrier);
-            $ganalyticsDataHandler->manageData('MBG.addCheckoutOption(2,\'' . $carrierName . '\');', 'A');
+
+            if ((bool) Configuration::get('GA_V4_ENABLED')) {
+                $js = $this->getGoogleAnalytics4($carrierName);
+            } else {
+                $js = $this->getUniversalAnalytics($carrierName);
+            }
+            $ganalyticsDataHandler->manageData($js, 'A');
         }
     }
 
     /**
-     * setParams
-     *
      * @param array $params
      */
     public function setParams($params)
     {
         $this->params = $params;
     }
+
+    protected function getUniversalAnalytics(string $carrierName)
+    {
+        return 'gtag("event", "add_shipping_info", {
+            currency: "' . $this->context->currency->iso_code . '",
+            value: ' . $this->context->cart->getCartTotalPrice() . ',
+            shipping_tier: "' . $carrierName . '"
+          });';
+    }
+
+    protected function getGoogleAnalytics4(string $carrierName)
```

Comment:
function name and inner content mismatch. In need of a content swapping.

Reply:
Fixed